cdc/processor/pipeline/table_actor.go

```diff
@@ -0,0 +1,400 @@
+				log.Error("failed to process message, stop table actor ",
+					zap.String("tableName", t.tableName),
+					zap.Int64("tableID", t.tableID), zap.Error(err))
+				t.stop(ctx, err)
+				break
+			}
+		}
+	}
+	// Report error to processor if there is any.
+	t.checkError()
+	return !t.stopped
+}
+
+func (t *tableActor) start(ctx context.Context) error {
+	if t.started {
+		log.Panic("start an already started table",
+			zap.String("changefeedID", t.changefeedID),
+			zap.Int64("tableID", t.tableID),
+			zap.String("tableName", t.tableName))
+	}
+	log.Debug("creating table flow controller",
+		zap.String("changefeedID", t.changefeedID),
+		zap.Int64("tableID", t.tableID),
+		zap.String("tableName", t.tableName),
+		zap.Uint64("quota", t.memoryQuota))
+
+	pullerNode := newPullerNode(t.tableID, t.replicaInfo, t.tableName, t.info.ID)
+	pCtx := NewContext(ctx, t.tableName, t.vars.TableActorSystem.Router(), t.actorID, t.info, t.vars)
+	if err := pullerNode.Init(pCtx); err != nil {
+		log.Error("puller fails to start",
+			zap.String("tableName", t.tableName),
+			zap.Int64("tableID", t.tableID),
+			zap.Error(err))
+		return err
+	}
+	t.pullerNode = pullerNode
+
+	flowController := common.NewTableFlowController(t.memoryQuota)
+	sorterNode := newSorterNode(t.tableName, t.tableID,
+		t.replicaInfo.StartTs, flowController,
+		t.mounter, t.replConfig,
+	)
+	sCtx := NewContext(ctx, t.tableName, t.vars.TableActorSystem.Router(), t.actorID, t.info, t.vars)
+	if err := sorterNode.StartActorNode(sCtx, true, t.wg); err != nil {
+		log.Error("sorter fails to start",
+			zap.String("tableName", t.tableName),
+			zap.Int64("tableID", t.tableID),
+			zap.Error(err))
+		return err
+	}
+	t.sortNode = sorterNode
+	var c AsyncMessageHolderFunc = func() *pipeline.Message { return pCtx.tryGetProcessedMessage() }
+	var d AsyncMessageProcessorFunc = func(ctx context.Context, msg pipeline.Message) (bool, error) {
+		return sorterNode.TryHandleDataMessage(sCtx, msg)
+	}
+	t.nodes = append(t.nodes, NewActorNode(c, d))
+
+	var cCtx *cyclicNodeContext
+	if t.cyclicEnabled {
+		cyclicNode := newCyclicMarkNode(t.markTableID)
+		cCtx = NewCyclicNodeContext(
+			NewContext(ctx, t.tableName, t.vars.TableActorSystem.Router(), t.actorID, t.info, t.vars))
+		if err := cyclicNode.Init(cCtx); err != nil {
+			log.Error("sink fails to start",
+				zap.String("tableName", t.tableName),
+				zap.Int64("tableID", t.tableID),
+				zap.Error(err))
+			return err
+		}
+		c = func() *pipeline.Message {
+			return sCtx.tryGetProcessedMessage()
+		}
+		d = func(ctx context.Context, msg pipeline.Message) (bool, error) {
+			return cyclicNode.TryHandleDataMessage(cCtx, msg)
+		}
+		t.nodes = append(t.nodes, NewActorNode(c, d))
+	}
+
+	actorSinkNode := newSinkNode(t.tableID, t.sink, t.replicaInfo.StartTs, t.targetTs, flowController)
+	if err := actorSinkNode.InitWithReplicaConfig(true, t.replConfig); err != nil {
+		log.Error("sink fails to start",
+			zap.String("tableName", t.tableName),
```

Comment:
// AsyncStop tells the pipeline to stop, and returns true if the pipeline is already stopped.

Reply:
done